test/e2e/commatrix_suite_test.go

```diff
@@ -108,8 +152,93 @@ var _ = AfterSuite(func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resp.StatusCode).ToNot(Equal(http.StatusNotFound))
+		}
+
+		By("write documented commatrix to artifact file")
+		docCommatrixContent, err := io.ReadAll(resp.Body)
+		Expect(err).ToNot(HaveOccurred())
+		docFilePath := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		err = os.WriteFile(docFilePath, docCommatrixContent, 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Filter documented commatrix for diff generation")
+		// get origin documented commatrix details
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, docFilePath, types.FormatCSV, infra, deployment)
+		Expect(err).ToNot(HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		Expect(err).ToNot(HaveOccurred())
+
+		if isSNO {
+			// Exclude all worker nodes static entries.
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: docComDetailsList}, "worker")
+			// Exclude static entries of standard deployment type.
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.StandardStaticEntries}, "master")
+		} else {
+			// Exclude specific master entries (see StandardExcludedMasterComDetails var description)
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: StandardExcludedMasterComDetails}, "master")
+		}
+
+		// if cluster is running on BM exclude Cloud static entries in diff generation
+		// else cluster is running on Cloud and exclude BM static entries in diff generation.
+		if isBM {
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.CloudStaticEntriesWorker}, "worker")
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.CloudStaticEntriesMaster}, "master")
+		} else {
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.BaremetalStaticEntriesWorker}, "worker")
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.BaremetalStaticEntriesMaster}, "master")
+		}
+		docComMatrix := &types.ComMatrix{Matrix: docComDetailsList}
+
+		By("generating diff between matrices for testing purposes")
+		diff := matrixdiff.Generate(commatrix, docComMatrix)
+		diffStr, err := diff.String()
+		Expect(err).ToNot(HaveOccurred())
+		err = os.WriteFile(filepath.Join(artifactsDir, "doc-diff-new"), []byte(diffFileComments+diffStr), 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("comparing new and documented commatrices")
+		// Get ports that are in the documented commatrix but not in the generated commatrix.
+		notUsedPortsMat := diff.GenerateUniqueSecondary()
+		if len(notUsedPortsMat.Matrix) > 0 {
+			logrus.Warningf("the following ports are documented but are not used: \n %s", notUsedPortsMat)
```

Comment:
Can you remove the space between \n and %s? There is an extra space in the first line when it's printed